manifests/plugin/supervisory_authority.pp

```diff
@@ -29,19 +29,34 @@
 # $metrics_interval::      Specify the interval for reporting metrics to APM Server.
 #
 class foreman::plugin::supervisory_authority (
-  Stdlib::HTTPUrl              $server_url,
-  String                       $secret_token,
+  Stdlib::HTTPUrl $server_url,
+  Variant[String, Sensitive[String]] $secret_token,
   Pattern[/^[a-zA-Z0-9 _-]+$/] $service_name,
-  Integer[0,5]                 $log_level             = 1,
-  Integer[0]                   $pool_size             = 1,
-  Integer[0]                   $api_buffer_size       = 256,
-  String                       $api_request_size      = '750kb',
-  String                       $api_request_time      = '10s',
-  Integer[0]                   $transaction_max_spans = 500,
-  Boolean                      $http_compression      = false,
-  String                       $metrics_interval      = '30s',
+  Integer[0,5] $log_level = 1,
+  Integer[0] $pool_size = 1,
+  Integer[0] $api_buffer_size = 256,
+  String $api_request_size = '750kb',
+  String $api_request_time = '10s',
+  Integer[0] $transaction_max_spans = 500,
+  Boolean $http_compression = false,
+  String $metrics_interval = '30s',
 ) {
   foreman::plugin { 'supervisory_authority':
-    config => template('foreman/foreman_supervisory_authority.yaml.erb'),
+    config => epp(
```

Comment:
Given there's a secret in here, should this itself be marked as Sensitive?
```suggestion
    config => Sensitive(epp(
```

Reply:
You are right.